### PR TITLE
task/KBDEV-1194 get updated studies for last two weeks in clinicaltrials

### PR DESCRIPTION
**Pull request by sshugsc**
[KBDEV-1194](https://www.bcgsc.ca/jira/browse/KBDEV-1194)
- update clinicaltrialsgov to load last updated data
- add maxRecords options to clinicaltrialsgov
- remove the out-of-dated rss/xml related code

### CODE REVIEW

File: src/clinicaltrialsgov/index.js [hunk 1]
```diff
@@ -391,13 +371,20 @@ const upload = async ({ conn }) => {
...
         });
 
         for (const trial of trials.studies) {
+            if (processCount > total) {
+                break;
+            }
+
             try {
                 const record = convertAPIRecord(trial);
+                logger.info(`processing (${processCount++}/${total}) record: ${record.sourceId}`);
```
Comment (mathieulemieux) [inline src/clinicaltrialsgov/index.js:387]: I think we shoul increment processCount on a seperate line.

File: src/clinicaltrialsgov/index.js [hunk 2]
```diff
@@ -366,9 +329,23 @@ const upload = async ({ conn }) => {
...
+    let processCount = 1,
+        total;
+
+    if (maxRecords) {
+        total = maxRecords;
+    } else {
+        total = trials.totalCount;
+    }
+
     for (const trial of trials.studies) {
```
Comment (mathieulemieux) [inline src/clinicaltrialsgov/index.js:341]: Why a loop here and then the same loop again at line 380?

**Review by mathieulemieux — CHANGES_REQUESTED** [review]

File: src/clinicaltrialsgov/README.md [hunk 3]
```diff
@@ -5,8 +5,19 @@ This module loads clinical trials data into GraphKB from [https://www.clinicaltr
...
 
 ```bash
-node bin/load.js api clinicaltrialsgov
+node bin/load.js clinicaltrialsgov
 ```
+
+Using `--maxRecords` can specify the maximum number of loaded studies. 
+```bash
+node bin/load.js --maxRecords 100 clinicaltrialsgov
+```
```
[the remark this answers is not in the document]
Comment (sshugsc) [reply src/clinicaltrialsgov/README.md]: Removed.

File: src/clinicaltrialsgov/index.js [hunk 4]
```diff
@@ -25,15 +14,13 @@ const {
 } = require('../graphkb');
 const { logger } = require('../logging');
 const { clinicalTrialsGov: SOURCE_DEFN } = require('../sources');
-const { api: apiSpec, rss: rssSpec } = require('./specs.json');
+const { api: apiSpec } = require('./specs.json');
```
[the remark this answers is not in the document]
Comment (sshugsc) [reply src/clinicaltrialsgov/index.js]: Updated.

File: src/clinicaltrialsgov/index.js (hunk 2 above)
Comment (sshugsc) [reply src/clinicaltrialsgov/index.js]: Put them in one loop now.

File: src/clinicaltrialsgov/index.js (hunk 1 above)
Comment (sshugsc) [reply src/clinicaltrialsgov/index.js]: Applied.

**Review by mathieulemieux — APPROVED** [review]

**Comment by mathieulemieux** [comment]
@sshugsc, to be safe, testing with some other loaders via the CLI would be a must before merging.

**Comment by mathieulemieux** [comment]
@sshu, can you post here which loaders you tested (the command you used) so we may want to use the same ones for testing the CLI from now on? I'm curious; have you choose one file loader, one api, cosmic and civic, in conjunction with the maxRecords flag?

**Comment by sshugsc** [comment]
> @sshu, can you post here which loaders you tested (the command you used) so we may want to use the same ones for testing the CLI from now on? I'm curious; have you choose one file loader, one api, cosmic and civic, in conjunction with the maxRecords flag?

I tested file loaders(ncit, refseq), api loaders(oncotree, docm), cosmic (resistance) and civic. They all work fine.

**Comment by mathieulemieux** [comment]
oh, that's a lot! Not the whole datasets I suppose; have you used the maxRecords flag?

**Comment by sshugsc** [comment]
> oh, that's a lot! Not the whole datasets I suppose; have you used the maxRecords flag?

yeah, tried maxRecords flag for them. Work good on most of them, but oncotree didn't support it.

**Pull request merged**